resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -60,4 +63,15 @@ private[spark] object KubernetesUtils {
   }
 
   def parseMasterUrl(url: String): String = url.substring("k8s://".length)
+
+  def buildPodWithServiceAccount(serviceAccount: Option[String], pod: SparkPod): Option[Pod] = {
```

Comment:
nit: would call this operation "add service account"

Reply:
Can the `pod` parameter here be null, or empty?

Reply:
I changed this to addServiceAccount but this differs from the accepted patch in 3.0.0
https://github.com/apache/spark/pull/24748/files

Reply:
sorry, I forgot that this was being pulled back to 2.4 as a bug-fix. Should probably keep it consistent w/ 3.0